### DISCUSSION TITLE
Adding --output flag

### DISCUSSION THREAD

**Discussion by baal-lgln** [citygml4j/citygml-tools#51] [Ideas]
Currently only the Input files can be set. It would be really helpful to set a different output directory like
`citygml-tools to-cityjson *.gml --output /cityjson`

**Comment by clausnagel** [comment]
I have two follow-up questions.

1. `to-cityjson` also supports to provide an entire directory as input. The directory is recursively scanned for CityGML input files. So, the input directory may contain nested directories of arbitrary depth. Would you expect the same directory structure to be created under the provided output directory for the CityJSON files?
2. What about references to texture files? Assume you have nested directories with CityGML files as described above, and for each CityGML file there is a additional relative folder where texture files are stored for this file. In case you create the CityJSON files at a different output directory, how should references to the texture files be handled: 1) the references should point to the texture files at the original input directory, or 2) the relative texture folders should also be copied to the new output directory so that the references remain local? What would you expect?

Point 2 (and especially 2.2) adds quite some complexity to the process. And copying texture files takes time and requires (possibly a lot of) additional disk space. That's why the `to-cityjson` command currently does not support specifying an output directory.

**Reply by baal-lgln** [reply]
Thank you for the fast response.
For 1. I think I would expect the same directory structure in the output.

For my current workflow I only use simple models without texture, so I haven't thought about that. But I would expect the textures to be copied to keep the reference local. But maybe other users would think differently about that. Maybe it would be possible to give the users some control over this with additional configuration options? But this would of course increase the complexity of the command.

**Comment by clausnagel** [comment]
Added with b17f10fa82eb27ddd96c6e9136760e52b4f33eb6. Will be included in the upcoming 2.4.0 release.
